test.sh

```diff
@@ -2,16 +2,35 @@
 
 set -eu
 
+xreadlink() {
+   # This sort of implements readlink -f except it works on OSX too
+
+   node=$1
+
+   cd `dirname $node`
+   node=`basename $node`
+
+   while [ -L "$node" ]
+   do
+       node=`readlink $node`
+
+       cd `dirname $node`
+       node=`basename $node`
+   done
+
+   echo `pwd`/$node
```

Comment:
Heh, this won't work if there are IFS characters other than space in the path... not sure if any filesystems support this...